webui/react/src/components/RunActionDropdown.tsx

```diff
@@ -0,0 +1,190 @@
+import { GridCell } from '@glideapps/glide-data-grid';
+import { ContextMenuCompleteHandlerProps } from 'hew/DataGrid/contextMenu';
+import Dropdown, { DropdownEvent, MenuItem } from 'hew/Dropdown';
+import { useModal } from 'hew/Modal';
+import { useToast } from 'hew/Toast';
+import useConfirm from 'hew/useConfirm';
+import { copyToClipboard } from 'hew/utils/functions';
+import React, { MouseEvent, useCallback, useMemo } from 'react';
+
+import usePermissions from 'hooks/usePermissions';
+import FlatRunMoveModalComponent from 'pages/FlatRuns/FlatRunMoveModal';
+import { handlePath } from 'routes/utils';
+import { archiveRuns, deleteRuns, killRuns, unarchiveRuns } from 'services/api';
+import { FlatRun, FlatRunAction, ValueOf } from 'types';
+import handleError, { ErrorLevel, ErrorType } from 'utils/error';
+import { getActionsForFlatRun } from 'utils/flatRun';
+import { capitalize } from 'utils/string';
+
+import { FilterFormSetWithoutId } from './FilterForm/components/type';
+
+interface Props {
+  children?: React.ReactNode;
+  cell?: GridCell;
+  run: FlatRun;
+  link?: string;
+  makeOpen?: boolean;
+  onComplete?: ContextMenuCompleteHandlerProps<FlatRunAction, void>;
+  onLink?: () => void;
+  onVisibleChange?: (visible: boolean) => void;
+  projectId: number;
+  filterFormSetWithoutId: FilterFormSetWithoutId;
+}
+
+const Action = {
+  Copy: 'Copy Value',
+  NewTab: 'Open Link in New Tab',
+  NewWindow: 'Open Link in New Window',
+  ...FlatRunAction,
+};
+
+type Action = ValueOf<typeof Action>;
+
+const dropdownActions = [Action.Archive, Action.Unarchive, Action.Kill, Action.Move, Action.Delete];
+
+const RunActionDropdown: React.FC<Props> = ({
+  run,
+  cell,
+  link,
+  makeOpen,
+  onComplete,
+  onLink,
+  onVisibleChange,
+  filterFormSetWithoutId,
+  projectId,
+}: Props) => {
+  const id = run.id;
```

Comment:
can we directly use `run.id` instead of `id`? i feel like id sounds like a too common word in this component because there are run id, project id etc. i think explicitly using `run.id` is better to understand what the `id` is